plugins/chat/app/queries/chat/users_from_usernames_and_groups.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Chat
+  class UsersFromUsernamesAndGroups
+    def self.call(usernames:, groups:)
+      User
+        .joins(:user_option)
+        .left_outer_joins(:groups)
+        .where(user_options: { chat_enabled: true })
+        .where(username: usernames)
+        .or(
+          User
+            .joins(:user_option)
+            .left_outer_joins(:groups)
+            .where(user_options: { chat_enabled: true })
+            .where(groups: { name: groups })
+            .where.not(group_users: { user_id: nil }),
+        )
+        .distinct
```

Comment:
Am I mistaken or this would be simpler (didn't try it):

```ruby
User
  .joins(:user_option)
  .left_outer_joins(:groups)
  .where(user_options: { chat_enabled: true })
  .where('username IN (?) OR (groups.name IN (?) AND group_users.user_id IS NOT NULL)', usernames, groups)
  .distinct
```